src/main/java/org/jabref/logic/importer/fileformat/EndnoteXmlImporter.java

```diff
@@ -125,366 +160,300 @@ public ParserResult importDatabase(BufferedReader input) throws IOException {
     }
 
-    private void parseRecord(XMLStreamReader reader, List<BibEntry> bibItems, String startElement)
-            throws XMLStreamException {
-
-        Map<Field, String> fields = new HashMap<>();
-        EntryType entryType = StandardEntryType.Article;
-
-        KeywordList keywordList = new KeywordList();
-        List<LinkedFile> linkedFiles = new ArrayList<>();
+    private BibEntry parseRecord(XMLStreamReader reader) throws XMLStreamException {
+        BibEntry entry = new BibEntry();
 
         while (reader.hasNext()) {
             reader.next();
-            if (isStartXMLEvent(reader)) {
+            if (isEndElement(reader, "record")) {
+                break;
+            }
+
+            if (isStartElement(reader)) {
                 String elementName = reader.getName().getLocalPart();
                 switch (elementName) {
                     case "ref-type" -> {
-                        String type = reader.getAttributeValue(null, "name");
-                        entryType = convertRefNameToType(type);
+                        String refType = reader.getAttributeValue(null, "name");
+                        EntryType entryType = ENTRY_TYPE_MAPPING.entrySet().stream()
+                                                                .filter(e -> e.getValue().equals(refType))
+                                                                .map(Map.Entry::getKey)
+                                                                .findFirst()
+                                                                .orElse(StandardEntryType.Misc);
+                        entry.setType(entryType);
                     }
                     case "contributors" -> {
-                        handleAuthorList(reader, fields, elementName);
+                        parseContributors(reader, entry);
                     }
                     case "titles" -> {
-                        handleTitles(reader, fields, elementName);
-                    }
-                    case "pages" -> {
-                        parseStyleContent(reader, fields, StandardField.PAGES, elementName);
-                    }
-                    case "volume" -> {
-                        parseStyleContent(reader, fields, StandardField.VOLUME, elementName);
-                    }
-                    case "number" -> {
-                        parseStyleContent(reader, fields, StandardField.NUMBER, elementName);
-                    }
-                    case "dates" -> {
-                        parseYear(reader, fields);
-                    }
-                    case "notes" -> {
-                        parseStyleContent(reader, fields, StandardField.NOTE, elementName);
+                        parseTitles(reader, entry);
                     }
-                    case "urls" -> {
-                        handleUrlList(reader, fields, linkedFiles);
+                    case "periodical", "alt-periodical" -> {
```

Comment:
With this, the order `periodical` and `alt-periodical` matters. Just leave out `alt-periodical` - or store it in the field `alt-periodical` (and keep it if it differs from `periodical`).

Reply:
Ah, found it! It was getting imported in the default case. Fixed it.